SGoopas/Assets/Scripts/MasterMonoBehaviour.cs

```diff
@@ -10,13 +11,37 @@ public class MasterMonoBehaviour : MonoBehaviour {
     public Text progressTxt;
 	public GameObject messageContainer;
 	private TextAnimator animator;
-
+	public GameObject imageHolder;
+	private bool isFading = false;
+	private float fadeTarget = 0f;
+	private Image fader;
+	public float fadeSpeed;
+	private Action faderCallback;
+	public GameObject pauseMenu;
     void Awake()
     {
         Instance = this;
 		animator = messageContainer.GetComponent<TextAnimator> ();
+		fader = imageHolder.GetComponent<Image> ();
+		pauseMenu.SetActive (false);
     }
 
+	void Update(){
+		if (isFading) {
+			float fadeChange = (fadeSpeed * Time.deltaTime) * Mathf.Sign(fadeTarget - fader.color.a);
```

Comment:
Can you move this logic to a separate class? I don't like the pattern of having a boolean for things in the update loop. i.e. create a Fader class that performs the fading and the callback, and propagate Update() down to that class. That way you can "Apply" the fader here instead of setting an arbitrary boolean.